lib/Doctrine/Migrations/Tools/Console/Command/UpToDateCommand.php

```diff
@@ -43,12 +45,26 @@ public function execute(InputInterface $input, OutputInterface $output) : ?int
             return 0;
         }
 
+        if ($availableMigrations > 0) {
+            $output->writeln(sprintf(
+                '<error>Out-of-date! %u migration%s available to execute.</error>',
+                $availableMigrations,
+                $availableMigrations > 1 ? 's are' : ' is'
+            ));
+
+            return 1;
+        }
+
+        // negative number means that there are unregistered migrations in the database
+
+        $extraMigrations = -$availableMigrations;
         $output->writeln(sprintf(
-            '<comment>Out-of-date! %u migration%s available to execute.</comment>',
-            $availableMigrations,
-            $availableMigrations > 1 ? 's are' : ' is'
+            '<error>You have %1$u previously executed migration%3$s in the database that %2$s registered migration%3$s.</error>',
+            $extraMigrations,
+            $extraMigrations > 1 ? 'are not' : 'is not a',
+            $extraMigrations > 1 ? 's' : ''
         ));
 
-        return 1;
+        return true === $input->getOption('fail-on-unregistered') ? 2 : 0;
```

Comment:
i originally had `$input->getOption('fail-on-unregistered') ? 2 : 0` but then the code sniffer complained that the return value of getOption is mixed. i tried to fix it like this, but now the code style check complains about yoda style comparison. how should i do the comparison?

Reply:
`$input->getOption('fail-on-unregistered') === true`